packages/init/src/index.ts

```diff
@@ -18,8 +20,16 @@ export default function initializeInquirer(...args: string[]): Function | void {
     const packages = args;
     const includesDefaultPrefix = packages.includes(AUTO_PREFIX);
     const generateConfig = packages.includes(CONFIG_PREFIX);
-    if (packages.length === 0 || includesDefaultPrefix || generateConfig) {
-        return modifyHelperUtil('init', initGenerator, null, null, includesDefaultPrefix, generateConfig);
+    const genPathPrefix = packages.includes(PATH_PREFIX);
+
+    let generatePath: string;
+    if (genPathPrefix) {
+        const idx = packages.indexOf(PATH_PREFIX);
+        // Retrieve the path supplied along with --generate-path
+        generatePath = packages[idx + 1];
+    }
+    if (packages.length === 0 || includesDefaultPrefix || generateConfig || genPathPrefix) {
+        return modifyHelperUtil('init', initGenerator, null, null, includesDefaultPrefix, generateConfig, generatePath);
```

Comment:
What is the magic logic?

Reply:
Maybe you can clarify?

Reply:
I haven't updated the logic, added an argument that would hold a reference to the specified path.

Reply:
I think here is bug, using `packages.indexOf(PATH_PREFIX)` can be wrong, because you can use `PATH_PREFIX` multiple time in path, we need look at logic and fix it

Reply:
`indexOf()` returns the first occurence.

Reply:
but it is bug, please refactor this logic, it is buggy

Reply:
We can provide support for `--generate-path=path` syntax once we leverage the `arg-parser`.